### PR TITLE
Check app name availability

### PR DESCRIPTION
**Pull request by whiskeyjay**
#34

### CODE REVIEW

**Comment by PrashanthCorp** [comment]
So @Nturinski asked me to repro this. 
As a new user, I created a resource group, a new plan and then a new appservice. I too ran into a 'hang state'. I suspect iut could also be due to a plan creation or the app service creation, not just the app name check

File: src/webAppCreator.ts [hunk 1]
```diff
@@ -400,18 +401,54 @@ class WebsiteStep extends WebAppCreatorStepBase {
...
-
-                if (!value.match(/^[a-z0-9\-]{0,59}$/ig)) {
-                    return 'App name should be 1-60 characters long and can only include alphanumeric characters and hyphens.';
+        let siteName: string;
+        let siteNameOkay = false;
+
+        // Do not use the system default DNS server(s) because sometimes it can be hijacked by ISV.
+        // Some ISV resolves names that don't exist to their own search site, causing false positive results.
+        dns.setServers([
+            '204.79.195.43', // prd1.azuredns-cloud.net
```
[the remark this answers is not in the document]
Comment (whiskeyjay) [reply src/webAppCreator.ts]: These are from public domain records for azurewebsites.net, prod1 ~ prod4.azuredns-cloud.net are listed as name servers for azurewebsites.net.

File: src/webAppCreator.ts [hunk 2]
```diff
@@ -426,23 +417,12 @@ class WebsiteStep extends WebAppCreatorStepBase {
...
-                dns.resolve(fullSiteName, (err, addresses) => {
-                    if (err && err.code === dns.NOTFOUND) {
-                        resolve(true);
-                        return;
-                    }
-
-                    resolve(false);
-                });
-            });
+            const nameAvailability = await client.checkNameAvailability(siteName, 'site');
```
Comment (EricJizbaMSFT) [inline src/webAppCreator.ts]: 😂👍😂

Comment (nturinski) [reply src/webAppCreator.ts]: Jay, be kind to yourself.  This is an awesome feature!

**Pull request merged**